pilot/pkg/model/service.go

```diff
@@ -653,3 +654,37 @@ func (s *Service) GetServiceAddressForProxy(node *Proxy) string {
 	}
 	return s.Address
 }
+
+// DeepCopy creates a clone of Service.
+// TODO : See if there is any efficient alternative to this function - copystructure can not be used as is because
+// Service has sync.RWMutex that can not be copied.
+func (s *Service) DeepCopy() *Service {
+	attrs := copyInternal(s.Attributes)
+	ports := copyInternal(s.Ports)
+	accounts := copyInternal(s.ServiceAccounts)
+	clusterVIPs := copyInternal(s.ClusterVIPs)
+
+	return &Service{
+		Attributes:      attrs.(ServiceAttributes),
+		Ports:           ports.(PortList),
+		ServiceAccounts: accounts.([]string),
+		CreationTime:    s.CreationTime,
+		Hostname:        s.Hostname,
+		Address:         s.Address,
+		ClusterVIPs:     clusterVIPs.(map[string]string),
+		Resolution:      s.Resolution,
+		MeshExternal:    s.MeshExternal,
+	}
+}
+
+func copyInternal(v interface{}) interface{} {
```

Comment:
Is there a benchmark of `copystructure`? Is that more efficient compared to copy one by one?

Reply:
I do not know of any benchmark of copy structure. I

Reply:
Do you prefer not using copystructure and just copy every thing ourselves?

Reply:
I prefer the more efficient way.

Reply:
Ok. I will go with this now and run a test and see if it is better to copy individuals. I will do a follow-up if it turns out individual copy is better